### PR TITLE
Unit Test LibPerpetuals

### PR DESCRIPTION
**Pull request by raymogg**
# Motivation
Adds unit testing for all branches in lib perpetuals.

# Changes
- adds a mock lib perpetuals contract
- adds unit tests

### CODE REVIEW

**Review by kumar-ish — APPROVED** [review]
:surfer:

File: contracts/test/LibPerpetualsMock.sol [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import "../lib/LibPerpetuals.sol";
+
+contract LibPerpetualsMock {
+    function canMatch(
+        Perpetuals.Order calldata a,
+        uint256 aFilled,
+        Perpetuals.Order calldata b,
+        uint256 bFilled
+    ) public view returns (bool) {
```
Comment (sporejack) [inline contracts/test/LibPerpetualsMock.sol]: Seeking clarification here on whether these should be declared `external` or `public`?

**Review by sporejack — APPROVED** [review]
Approved; left a comment for discussion.

File: contracts/test/LibPerpetualsMock.sol (hunk 1 above)
Comment (CalabashSquash) [reply contracts/test/LibPerpetualsMock.sol]: Would say it doesn't matter since they are just used for testing

**Review by CalabashSquash — APPROVED** [review]
:morocco:

File: contracts/test/LibPerpetualsMock.sol (hunk 1 above)
Comment (raymogg) [reply contracts/test/LibPerpetualsMock.sol]: @sporejack probably should be external, I'll change it just for consistency. But as Ben mentioned, this is just for testing so its not too critical.

**Pull request merged**